Added env var to enable WS connections

aalu1418 opened the pull request:
Added environment variable `WS_ENABLED` that needs to be set for EAs to use WS connections (if available).

RodrigoAD commented on packages/core/bootstrap/src/index.ts:
```diff
@@ -139,10 +139,11 @@ const executeSync = (execute: Execute, makeWsHandler?: MakeWSHandler): ExecuteSy
-  ].concat(metrics.METRICS_ENABLED ? [withMetrics, withDebug] : [withDebug])
+  ]
+    .concat(metrics.METRICS_ENABLED ? [withMetrics, withDebug] : [withDebug])
+    .concat(ws.WS_ENABLED ? [ws.withWebSockets(storeSlice('ws'), makeWsHandler)] : [])
```
You can check it inside the ws middleware better, so it would be like:
```js
if (!makeWsHandler || !WS_ENABLED) return await execute(input) // Middleware is ignored
// Middleware continues
```

krebernisak replied:
The order of these middlewares matters, so my suggestion is to move it back in its place and make the check inside the middleware. Also agree that the enable flag should be set internally in WS config.

RodrigoAD commented on packages/core/bootstrap/src/lib/ws/index.ts:
```diff
@@ -3,13 +3,16 @@ import { Store } from 'redux'
 export * as reducer from './reducer'
 export * as epics from './epics'
 
+export const WS_ENABLED = parseBool(process.env.WS_ENABLED)
```
You can include it inside `ws/config.ts` and export it through `getWSConfig`

boxhock commented:
Has this been requested by anyone? Any thoughts on explicitly disabling it instead? Right now I can't think of any cases where node ops _don't_ want to use WS.

justinkaseman commented:
> Has this been requested by anyone? Any thoughts on explicitly disabling it instead? Right now I can't think of any cases where node ops _don't_ want to use WS.

I requested this as I would like a safety hatch in case this rollout of WS adapters goes bad. 

We should keep WS on by default, but allow people to disable.

boxhock commented:
@justinkaseman perfect, thanks that makes sense

aalu1418 commented:
Updates:
* Rebased with changes from over the weekend
* Moved the check inside of `ws/index` as suggested
* Moved/modified old `WS_ENABLED` parameter in `ws/config`
  * Removed `experimental` from environment variable
  * Moved enable toggle within `wsConfig` (not used anywhere else as far as I could tell)
